torchaudio/io/_compat.py

```diff
@@ -116,3 +117,151 @@ def load_audio_fileobj(
 ) -> Tuple[torch.Tensor, int]:
     s = torchaudio.lib._torchaudio_ffmpeg.StreamReaderFileObj(src, format, None, buffer_size)
     return _load_audio(s, frame_offset, num_frames, convert, channels_first)
+
+
+def _get_sample_format(dtype: torch.dtype) -> str:
+    dtype_to_format = {
+        torch.uint8: "u8",
+        torch.int16: "s16",
+        torch.int32: "s32",
+        torch.int64: "s64",
+        torch.float32: "flt",
+        torch.float64: "dbl",
+    }
+    if format := dtype_to_format.get(dtype):
+        return format
+    else:
+        raise ValueError(f"No format found for dtype {dtype}; dtype must be one of {list(dtype_to_format.keys())}.")
+
+
+def _native_endianness():
+    if sys.byteorder == "little":
+        return "le"
+    else:
+        return "be"
+
+
+def _get_encoder_for_wav(dtype, encoding, bits_per_sample):
+    assert bits_per_sample in {None, 8, 16, 24, 32, 64}
```

Comment:
`assert` does not run in optimized mode, (so in general `assert` should be used for validating logic internal to implementation) can we replace this with `if ~ raise` pattern?